cookbooks/aws-parallelcluster-environment/spec/unit/recipes/config_default_user_home_spec.rb

```diff
@@ -29,11 +29,13 @@
           expect(chef_run).to run_bash("Verify data integrity for #{user_home}").with(
             code: <<-CODE
     diff_output=$(diff -r #{user_home} #{user_local_home})
-    if [ $? -eq 0 ]; then
+    if [[ $diff_output != *"Only in #{user_home}"* ]]; then
       rm -rf /tmp#{user_home}
       rm -rf #{user_home}
```

Comment:
Also I'm just noticing this `rm -rf #{user_home}`. This is very risky and should never be done.
We should only delete temporary folders.

Reply:
We had a new `user_local_home`. Why we still need `user_home` which has the data `user_local_home` already have? This `rm -rf #{user_home}` is the existing logic before I created the data integrity check PR. And we have used it for a long time. So it should be okay.

Also, it's not a recipe, it's a spec test.

My concern is maybe the temp folder is useless. In this case, it's not like the `restore_home_shared_data.rb`, in `restore_home_shared_data.rb` we replace /home dir, so we need a temp folder to store data. But in this case, we can directly sync data from #{node['cluster']['cluster_user_home'] to #{node['cluster']['cluster_user_local_home']. But also the temp folder is the existing logic before I created the data integrity check PR.

Reply:
> We had a new user_local_home. Why we still need user_home which has the data user_local_home already have?

We should avoid to do `rm -rf` on home directories not because we may need the data in these directories, but because we do not want to take any risk of data loss. What if, for whatever race condition, a remote file system is still attached to the original home directory? In that case that rm -rf would remove everything from the remote file system and we would cause data loss.

We should reason in terms of cost/opportunity.
1. What the worst case cost of doing rm -rf? In the worst case it causes data loss, so potentially millions of dollars in data and reputation.
2. What's the cost of keeping the original home directory?  A potential confusion in the user that may not understand why the original home is kept. This would cost barely nothing to the cx and only potential tickets to us that could be resolved with documentation.

> Also, it's not a recipe, it's a spec test.

A spec test typically reflects what the recipe is doing. If it does reflect it, then my comment applies to both the test and the recipe. If the test does not reflect the recipe, then we have a problem in the test.

Reply:
Yes, that explain my confusion, thank you Giacomo. I totally agree with the potential risks you mentioned. 

> Furthermore, at some point we may reorder our recipes forgetting about the requirement for this recipe to be executed at the beginning, making the recursive deletion a potential source of data loss.

You are correct! Can not agree more.

> A spec test typically reflects what the recipe is doing. If it does reflect it, then my comment applies to both the test and the recipe. If the test does not reflect the recipe, then we have a problem in the test.

I know! I was just trying to say we should look at the recipe.

Reply:
Done and add a comment for it.